app/presenters/tree_node_builder.rb

```diff
@@ -24,18 +24,18 @@ def self.generic_tree_node(key, text, image, tip = nil, options = {})
   #   :open_nodes -- Tree node ids of currently open nodes
   #   FIXME: fill in missing docs
   #
-  def self.build(object, parent_id, options)
-    builder = new(object, parent_id, options)
+  def self.build(object, parent_id, options, sandbox)
+    builder = new(object, parent_id, options, sandbox)
     builder.build
   end
 
-  def self.build_id(object, parent_id, options)
-    builder = new(object, parent_id, options)
+  def self.build_id(object, parent_id, options, sandbox)
+    builder = new(object, parent_id, options, sandbox)
     builder.build_id
   end
 
-  def initialize(object, parent_id, options)
-    @object, @parent_id, @options = object, parent_id, options
+  def initialize(object, parent_id, options, sandbox)
+    @object, @parent_id, @options, @sb = object, parent_id, options, sandbox
```

Comment:
@h-kataria : what about the `@sb`? What about adding a new single purpose key to @options instead of adding a new argument and passing the whole `@sb`?